apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/transaction/messages/AbstractAttachment.java

```diff
@@ -56,6 +56,7 @@ public String getAppendixName() {
 
     @Override
     public void validate(Transaction transaction, int blockHeight) throws AplException.ValidationException {
+        this.transactionType = transaction.getType();
```

Comment:
Why should we set something in side method validate?

Reply:
Because code calls 
public void validate(Transaction transaction...
**transactionType()**.validateAttachment(....

but that is checks :
private TransactionType transactionType() {
        if (**this.transactionType == null**) {
            throw new IllegalStateException("Transaction type was not set");

and then throws IllegalStateException

That is current error on local PC if you try to start forging.

Reply:
I guess it's not a good solution.
We need to set the trunsactionType either in the constructor or using the bindTransactionType method